common/web/lm-worker/src/test/mocha/cases/suggestion-finalization.js

```diff
@@ -0,0 +1,418 @@
+import { DummyModel } from "#./models/dummy-model.js";
+
+/*
+ * This file's tests use these parts of a lexical model:
+ * - model.wordbreaker
+ * - model.punctuation
+ */
+
+const testModelWithSpacing = new DummyModel({
+  wordbreaker: wordBreakers.default,
+  punctuation: {
+    quotesForKeepSuggestion: {
+      open: '<',
+      close: '>'
+    },
+    insertAfterWord: '\u00a0' // non-breaking space
+  }
+});
+
+const testModelWithoutSpacing = new DummyModel({
+  wordbreaker: wordBreakers.default,
+  punctuation: {
+    quotesForKeepSuggestion: {
+      open: '<',
+      close: '>'
+    },
+    insertAfterWord: ''
+  }
+});
+
+/**
+ * Builds a fresh copy of test values useful for suggestion-similarity
+ * testing.
+ * @returns
+ */
+const build_its_is_set = () => {
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const its = {
+    correction: {
+      sample: 'its',
+      p: 0.8
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 's',
+          deleteLeft: 0
+        },
+        displayAs: 'its'
+      },
+      p: 0.2
+    },
+    totalProb: 0.16
+    // matchLevel does not yet exist.
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const it_is = {
+    correction: {
+      sample: 'its',
+      p: 0.8
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: '\'s',
+          deleteLeft: 0
+        },
+        displayAs: 'it\'s'
+      },
+      p: 0.8
+    },
+    totalProb: 0.64
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const is = {
+    correction: {
+      sample: 'is',
+      p: 0.2
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 's',
+          deleteLeft: 1
+        },
+        displayAs: 'is'
+      },
+      p: 0.5
+    },
+    totalProb: 0.1
+  };
+
+  /** @type {import('#./predict-helpers.js').CorrectionPredictionTuple} */
+  const is_not = {
+    correction: {
+      sample: 'is',
+      p: 0.2
+    },
+    prediction: {
+      sample: {
+        transform: {
+          insert: 'sn\'t',
+          deleteLeft: 1
+        },
+        displayAs: 'isn\'t'
+      },
+      p: 0.5
+    },
+    totalProb: 0.1
+  };
+
+  return {
+    its,
+    it_is,
+    is,
+    is_not
+  }
+};
+
+describe('finalizeSuggestions', () => {
+  describe('with custom post-token insert', () => {
+    it('with caret at end of current token and context', () => {
+      /** @type {Context} */
+      const context = {
+        left: 'It',
+        right: '',
+        startOfBuffer: true,
+        endOfBuffer: true
+      };
+
+      /** @type {ProbabilityMass<Transform>} */
+      const transform = {
+        insert: 's',
+        deleteLeft: 0
+      };
+
+      const testSet = build_its_is_set();
+      const analyzedSuggestions = [...Object.values(testSet).map((entry) => deepCopy(entry))];
```

Comment:
This pattern + `expected` is repeated throughout. Can we refactor?

```suggestion
      const { analyzedSuggestions, testSet, expected } = build_its_is_set();
```

and in `build_its_is_set` return all of these? `expected` seems to be identical in 8 of 9 tests.